dex/networks/erc20/contracts/ERC20SwapV0.sol

```diff
@@ -0,0 +1,207 @@
+// SPDX-License-Identifier: BlueOak-1.0.0
+// pragma should be as specific as possible to allow easier validation.
+pragma solidity = 0.8.6;
+
+contract ERC20Swap {
+    bytes4 private constant TRANSFER_FROM_SELECTOR = bytes4(keccak256("transferFrom(address,address,uint256)"));
+    bytes4 private constant TRANSFER_SELECTOR = bytes4(keccak256("transfer(address,uint256)"));
+
+    // State is a type that hold's a contract's state. Empty is the uninitiated
+    // or null value.
+    enum State { Empty, Filled, Redeemed, Refunded }
+
+    // Swap holds information related to one side of a single swap. The order of
+    // the struct fields is important to efficiently pack the struct into as few
+    // 256-bit slots as possible to reduce gas cost. In particular, the 160-bit
+    // address can pack with the 8-bit State.
+    struct Swap {
+        bytes32 secret;
+        uint256 value;
+        uint initBlockNumber;
+        uint refundBlockTimestamp;
+        address initiator;
+        address participant;
+        address token;
```

Comment:
Is `token` in actuality the address of a contract?

Reply:
Yes.